### PR TITLE
Win32: Fix keymap for keyboard layouts that can print UTF-16 surrogates and ligatures

### PR DESCRIPTION
**Pull request by DJm00n**
Old implementation with `MapVirtualKey(..., MAPVK_VK_TO_CHAR) & 0x7FFFF` simply returned `A`..`Z` for VK_A..VK_Z and
completely useless <U+0002 START OF TEXT> (`WCH_LGTR 0xF002` without high-order bit) in case of ligature.

Keymap before/after:
- [Gothic keyboard](https://www.diffchecker.com/75lM72Pr/)
- [Russian keyboard](https://www.diffchecker.com/Q7A59q38/)

See https://kbdlayout.info/features/ligatures for a list of affected keyboard layouts with ligatures.

More info on `MAPVK_VK_TO_CHAR` behavior: https://stackoverflow.com/a/72464584/1795050

### CODE REVIEW

**Comment by DJm00n** [comment]
@slouken I can change the code to exclude this behavior change for VK_A..VK_Z if you want.

**Comment by slouken** [comment]
Let me know when your keyboard patches are ready to go in, and if there are any order dependencies? The scancode patch is failing CI, for example.

**Comment by DJm00n** [comment]
@slouken this one should be ready to go.

File: src/video/windows/SDL_windowskeyboard.c [hunk 1]
```diff
@@ -130,20 +128,44 @@ void WIN_UpdateKeymap(SDL_bool send_event)
...
+        //} else {
+        {
+            BYTE keyboardState[256] = { 0 };
+            WCHAR buffer[16] = { 0 };
+            Uint32 *ch = 0;
+            int result = ToUnicode(vk, i, keyboardState, buffer, 16, 0);
+            buffer[SDL_abs(result) + 1] = 0;
+
+            /* Convert UTF-16 to UTF-32/UCS-4 code points */
+            ch = (Uint32 *)SDL_iconv_string("UCS-4-INTERNAL", "UTF-16LE", (const char *)buffer, (SDL_abs(result) + 1) * sizeof(WCHAR));
```
Comment (slouken) [inline src/video/windows/SDL_windowskeyboard.c:154]: The -INTERNAL variants are deprecated, this should probably just use "UCS-4" or "UTF-32LE"

Comment (DJm00n) [reply src/video/windows/SDL_windowskeyboard.c]: fixed

**Pull request merged**